### PR TITLE
Add academic context to chosen schedule date.

### LINKED ISSUE #49: As a student, I want to view my calendar events with respect to NUS’ academic weeks so that I can better keep track of my tasks in tandem with the school semester.

### PR DESCRIPTION
**Pull request by limsiying**
Closes #49, #54, #70.

### CODE REVIEW

File: tutorials/javaFxTutorialPart3.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ import javafx.scene.image.ImageView;
 ```
 
 Next, add two images to the `main/resources/images` folder.
-For this tutorial, we have two images `DaUser.png` and `DaDuke.png` to represent the user avatar and duke's avatar respectively but you can use any image you want.
+For this tutorial, we have two images `DaUser.png` and `DaDuke.png` to represent the user avatar and duchess's avatar respectively but you can use any image you want.
```
Comment (muserr) [inline tutorials/javaFxTutorialPart3.md]: Shouldn't it be DaDuchess.png? 😂😂

File: src/main/java/duchess/model/AcademicContext.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            case 14:
+                currWeek--;
+                break;
+            case 15:
+                return "Reading Week";
+            case 16:
+            case 17:
+                return "Examinations";
+            default:
+                break;
```
Comment (muserr) [inline src/main/java/duchess/model/AcademicContext.java]: Not necessary to have a default here.

File: src/main/java/duchess/model/AcademicContext.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            }
+        } else {
+            if (currWeek == 1) {
+                return "last week of break before a new semester starts";
+            }
+            return currWeek + " weeks left to the next semester";
+        }
+        return "W" + currWeek;
```
Comment (muserr) [inline src/main/java/duchess/model/AcademicContext.java]: May want to use if-else if-else here for further clarity.

File: tutorials/javaFxTutorialPart3.md (hunk 1 above)
Comment (limsiying) [reply tutorials/javaFxTutorialPart3.md]: :O nani i didn't realise i changed this

File: src/main/java/duchess/model/AcademicContext.java (hunk 2 above)
Comment (limsiying) [reply src/main/java/duchess/model/AcademicContext.java]: my default is for weeks 1 - 6, have changed to `return "W" + currWeek;`.

File: src/main/java/duchess/model/AcademicContext.java (hunk 3 above)
Comment (limsiying) [reply src/main/java/duchess/model/AcademicContext.java]: have changed!

File: src/main/java/duchess/model/AcademicContext.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    private String processAcademicWeek() {
+        boolean isSchoolTerm = semester.equals("Semester 1") || semester.equals("Semester 2");
+
+        long daysBetween = ChronoUnit.DAYS.between(startDate, endDate);
+        if (isSchoolTerm) {
+            daysBetween++;
+        }
+        int currWeek = (int) Math.ceil(daysBetween / 7.0);
```
Comment (muserr) [inline src/main/java/duchess/model/AcademicContext.java]: Please use a static final variable for 7.0 here! Can avoid magic numbers 👍

File: src/main/java/duchess/model/AcademicContext.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            default:
+                return "W" + currWeek;
+            }
+        } else {
+            if (currWeek == 1) {
+                return "last week of break before a new semester starts";
+            } else {
+                return currWeek + " weeks left to the next semester";
+            }
+        }
```
Comment (muserr) [inline src/main/java/duchess/model/AcademicContext.java]: Deep nesting can be avoided here :)

File: src/main/java/duchess/model/AcademicContext.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    private String processAcademicYear() {
+        int currYear = localDate.getYear();
+        boolean isFirstHalf = semester.equals("Semester 1")
+                || (semester.equals("Winter Break") && localDate.getMonth() == Month.DECEMBER);
+
+        if (isFirstHalf) {
+            return "AY" + currYear + "/" + (currYear + 1);
+        }
+        return "AY" + (currYear - 1) + "/" + currYear;
```
Comment (muserr) [inline src/main/java/duchess/model/AcademicContext.java]: Can use if - else?

File: src/main/java/duchess/model/AcademicContext.java (hunk 4 above)
Comment (limsiying) [reply src/main/java/duchess/model/AcademicContext.java]: Changed 👍

File: src/main/java/duchess/model/AcademicContext.java (hunk 5 above)
Comment (limsiying) [reply src/main/java/duchess/model/AcademicContext.java]: Removed `else` nest.

File: src/main/java/duchess/model/AcademicContext.java (hunk 6 above)
Comment (limsiying) [reply src/main/java/duchess/model/AcademicContext.java]: Added else condition

**Pull request merged**